src/hotspot/share/cds/filemap.cpp

```diff
@@ -747,8 +747,7 @@ void FileMapInfo::open_as_output() {
   // Use remove() to delete the existing file because, on Unix, this will
   // allow processes that have it open continued access to the file.
   remove(_full_path);
-  int mode = CDSConfig::is_dumping_preimage_static_archive() ? 0666 : 0444;
-  int fd = os::open(_full_path, O_RDWR | O_CREAT | O_TRUNC | O_BINARY, mode);
+  int fd = os::open(_full_path, O_RDWR | O_CREAT | O_TRUNC | O_BINARY, 0666);
```

Comment:
Why `0666`, though, i.e. world-writable? None of the tools, AFAICS from your examples, do world-writability. Should be `0644`?

Reply:
Also matches what Linux does:

```
$ rm 1; touch 1; stat 1 | grep Access
Access: (0664/-rw-rw-r--)  Uid: ( 1000/   shade)   Gid: ( 1000/   shade)
```

Reply:
Oh, TIL.